bubble_sort_by.rb

```diff
@@ -0,0 +1,18 @@
+def bubble_sort_by(arr)
+  n = arr.length
+  loop do
+    swapped = false
+    (n - 1).times do |i|
+      if yield(arr[i], arr[i + 1]).positive?
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+        swapped = true
+      end
+    end
+    break unless swapped
+  end
+  arr
+end
+sorted_array = bubble_sort_by(%w[hey hello hi]) do |left, right|
+  left.length - right.length
+end
+puts sorted_array
```

Comment:
From line 15 to line 18 is debugging code. Please remove these lines. :hammer_and_wrench: :smile: